src/unity/python/turicreate/visualization/plot.py

```diff
@@ -126,33 +127,44 @@ def save(self, filepath, include_data=True):
         >>> plt.save('vega_spec.json', False)
 
         """
-        spec = _json.loads(self.__proxy__.get('call_function', {'__function_name__': 'get_spec'}))
-
-        if(include_data):
-            data = _json.loads(self.__proxy__.get('call_function', {'__function_name__': 'get_data'}))["data_spec"]
-
-            for x in range(0, len(spec["vega_spec"]["data"])):
-                if(spec["vega_spec"]["data"][x]["name"] == "source_2"):
-                    spec["vega_spec"]["data"][x] = data
-                    break;
+        if type(filepath) != str:
+            raise ValueError("File path provided is not a string")
+
+        if filepath.endswith(".json"):
+            # save as vega json
+            spec = self._get_vega(include_data = include_data)
+            with open(filepath, 'w') as fp:
+                _json.dump(spec, fp)
+        elif filepath.endswith(".png") or filepath.endswith(".svg"):
+            # save as png/svg, but json first
+            spec = self._get_vega(include_data = True)
+            extension = filepath[-3:]
+            with open("temp_file.vg.json", 'w') as fp:
+                _json.dump(spec, fp)
+            dirname = os.path.dirname(__file__)
+            relative_path_to_vg2png_vg2svg = "../../../../../../node_modules/vega/bin/vg2" + extension
+            absolute_path_to_vg2png_vg2svg = os.path.join(dirname, relative_path_to_vg2png_vg2svg)
+            os.system("node " + absolute_path_to_vg2png_vg2svg + " temp_file.vg.json " + filepath)
+            # delete temp file that user didn't ask for
+            os.system("rm temp_file.vg.json") 
+        else:
+            raise NotImplementedError("Plot save() is only supported for json, png, and svg")
 
-        with open(filepath, 'w') as fp:
-            _json.dump(spec, fp)
 
     def _get_data(self):
         return _json.loads(self.__proxy__.get('call_function', {'__function_name__': 'get_data'}))
 
     def _get_vega(self, include_data=True):
         if(include_data):
-            spec = _json.loads(self.__proxy__.get('call_function', {'__function_name__': 'get_spec'}))
+            spec = _json.loads(self.__proxy__.get('call_function', {'__function_name__': 'get_spec'}))["vega_spec"]
             data = _json.loads(self.__proxy__.get('call_function', {'__function_name__': 'get_data'}))["data_spec"]
-            for x in range(0, len(spec["vega_spec"]["data"])):
-                if(spec["vega_spec"]["data"][x]["name"] == "source_2"):
-                    spec["vega_spec"]["data"][x] = data
+            for x in range(0, len(spec["data"])):
+                if(spec["data"][x]["name"] == "source_2"):
```

Comment:
what is the significance of "source_2"?

Reply:
@abhishekpratapa ?

Reply:
`source_2` seems to be the default name for a dataset within Vega (if you don't otherwise give it a name) - so we've adopted it by convention for our dataset. We should probably rename it to something clearer (since we don't need to use the default for this), although that change would need to touch Python and JavaScript code and is orthogonal to this PR, so it could come later as a cleanup PR. For now let's keep `source_2`.